src/metabase/driver/materialize.clj

```diff
@@ -3,29 +3,66 @@
+  (let [adjusted-amount (if (= unit :week) (* 7 amount) amount)
+        adjusted-unit (if (= unit :week) :day unit)]
+    (h2x// (sql.qp/add-interval-honeysql-form :postgres hsql-form adjusted-amount adjusted-unit))))
+
+;;; +----------------------------------------------------------------------------------------------------------------+
+;;; |                                          metabase.driver method impls                                          |
+;;; +----------------------------------------------------------------------------------------------------------------+
+
+(doseq [[feature supported?] {:foreign-keys              (not config/is-test?)
+                              ;; Materialize defaults to UTC, and this is the only supported value
+                              :set-timezone              false
+                              :datetime-diff             false
+                              :convert-timezone          false
+                              :temporal-extract          (not config/is-test?)
+                              ;; Disabling nested queries during tests as they try to use Foreign Keys
+                              :nested-queries            (not config/is-test?)
+                              ;; Disabling the expressions support due to the following error:
+                              ;; Error executing query: ERROR: function substring(text, character varying) does not exist
+                              :expressions               false
+                              ;; Disabling model caching:
+                              :persist-models            false
+                              ;; Disable percentile aggregations due to missing support for PERCENTILE_CONT
+                              :percentile-aggregations   false
+                              :test/jvm-timezone-setting false}]
+  (defmethod driver/database-supports? [:materialize feature] [_driver _feature _db] supported?))
+
+(defmethod sql-jdbc.execute/set-timezone-sql :materialize
+  [_]
+  "SET TIMEZONE TO %s;")
+
 ; ;;; +----------------------------------------------------------------------------------------------------------------+
 ; ;;; |                                         metabase.driver.sql-jdbc impls                                         |
 ; ;;; +----------------------------------------------------------------------------------------------------------------+
 
 (defmethod sql-jdbc.conn/connection-details->spec :materialize
-  [_ {:keys [host port db], :as opts}]
+  [_ {:keys [host port db cluster], :as opts}]
   (sql-jdbc.common/handle-additional-options
    (merge
     {:classname                     "org.postgresql.Driver"
      :subprotocol                   "postgresql"
-     :subname                       (str "//" host ":" port "/" db)
-     :ssl                           true
+     :subname                       (str "//" host ":" port "/" db "?options=--cluster%3D" cluster)
```

Comment:
Just updated this, let me know if this is what you had in mind

Reply:
Given that `host`, `port`, etc. are user input, we cannot trust them to be safe. We construct a string from them and pass that to the driver. Since we probably cannot fully trust the driver either, we should do everything reasonable to make sure there is nothing suspicious/bad in that string.

Reply:
Do you have any suggestions for best practices?
I tried to check how other similar drivers handle this like Redshift [here](https://github.com/metabase/metabase/blob/master/modules/drivers/redshift/src/metabase/driver/redshift.clj#L277C1-L286) and some other partner drivers like [Exasol](https://github.com/exasol/metabase-driver/blob/main/src/metabase/driver/exasol.clj#L74C9-L87) and [ClickHouse](https://github.com/ClickHouse/metabase-clickhouse-driver/blob/master/src/metabase/driver/clickhouse.clj#L73-L90) but I belive that they also just construct a string and pass it to the driver.

Reply:
It depends on the underlying driver. I'm guessing that
- the resulting string should be a valid URI,
- `host` should be a host name or IP address, 
- `port` should be an integer,
- `db` should be a valid database name, and
- `cluster` should be a valid cluster name.

Reply:
Currently out of ideas on a good way to fulfill this request, as this seems to be the way that it's currently handled with all other drivers:

https://github.com/metabase/metabase/blob/master/src/metabase/db/spec.clj#L24-L29